optimade/models/utils.py

```diff
@@ -182,6 +183,22 @@ def base_version(self) -> str:
         return f"{self.major}.{self.minor}.{self.patch}"
 
 
+def anonymous_element_generator():
+    """ Generator that yields the next symbol in the A, B, Aa, ... Az naming scheme. """
+    from string import ascii_lowercase
+
+    for size in itertools.count(1):
+        for s in itertools.product(ascii_lowercase, repeat=size):
+            s = list(s)
+            s[0] = s[0].upper()
+            yield "".join(s)
+
+
+ANONYMOUS_ELEMENTS = tuple(itertools.islice(anonymous_element_generator(), 150))
+""" Returns the first 150 values of the anonymous element generator. """
+
+CHEMICAL_FORMULA_REGEXP = r"[A-Z][a-z]?\d*"
```

Comment:
I would like a better regexp than this, i.e. one that has to match the entire string and disallows `"Ag...Cl"` `" AgCl"` etc

Reply:
Fiddling with `^` and `$` and having no luck

Reply:
Was just missing a final qualifier...